SConstruct

```diff
@@ -165,6 +164,8 @@ def set_defaults(env, daos_version):
             env.Append(CCFLAGS=['-DDAOS_BUILD_RELEASE'])
         env.AppendUnique(CCFLAGS=['-O2', '-D_FORTIFY_SOURCE=2'])
 
+    if env["CC"] != 'icx':
+        DESIRED_FLAGS.extend(['-fno-delete-null-pointer-checks'])
```

Comment:
This should check if the option is supported and add if it it's not, the fact you're making it conditional on compiler version indicates that it turns on some checking that then subsequently fails, is that the case?

Also, the mixing of " and ' on one line throws me, can we stick on ' please?

Reply:
I should revisit it...this was added in a prior incarnation.   If I recall, the option checks never worked on icc because it doesn't fail

Reply:
I've updated to remove this change....not sure why I put that there initially